core/src/main/java/com/oath/oak/ValueUtils.java

```diff
@@ -0,0 +1,382 @@
+                                            OakSerializer<V> serializer, MemoryManager memoryManager) {
+        ByteBuffer bb = lookUp.valueSlice.getByteBuffer();
+        int capacity = serializer.calculateSize(newVal);
+        if (bb.remaining() < capacity + ValueUtils.VALUE_HEADER_SIZE) { // can not reuse the existing space
+            if (!chunk.publish()) {
+                return RETRY;
+            }
+            bb.putInt(bb.position(), MOVED.value);
+            ByteBuffer dup = bb.duplicate();
+            dup.position(dup.position() + ValueUtils.VALUE_HEADER_SIZE);
+            Slice s = lookUp.valueSlice;
+            Slice sDup = new Slice(s.getBlockID(), dup);
+            memoryManager.releaseSlice(sDup);
+            s = memoryManager.allocateSlice(capacity + VALUE_HEADER_SIZE);
+            lookUp.valueSlice = s;
+            bb = s.getByteBuffer();
+            bb.putInt(bb.position(), LOCKED.value);
+            int valueBlockAndLength =
+                    (s.getBlockID() << VALUE_BLOCK_SHIFT) | ((capacity + VALUE_HEADER_SIZE) & VALUE_LENGTH_MASK);
+            assert chunk.casEntriesArrayLong(lookUp.entryIndex, Chunk.OFFSET.VALUE_REFERENCE, lookUp.valueReference,
+                    UnsafeUtils.intsToLong(valueBlockAndLength, bb.position()));
+            chunk.unpublish();
+        }
+        ByteBuffer dup = getValueByteBufferNoHeader(bb);
+        // It seems like I have to change bb here or else I expose the header to the user
+        // Duplicating bb instead
+        serializer.serialize(newVal, dup);
+        return SUCCESS;
+    }
+
+    /**
+     * @param s        the value Slice (including the header)
+     * @param computer the function to apply on the Slice
+     * @return {@code SUCCESS} if the function was applied successfully,
+     * {@code FAILURE} if the value is deleted,
+     * {@code RETRY} if the value was moved.
+     */
+    static ValueResult compute(Slice s, Consumer<OakWBuffer> computer) {
+        ByteBuffer bb = s.getByteBuffer();
+        ValueResult res = lockWrite(bb);
+        if (res != SUCCESS) {
+            return res;
+        }
+        OakWBuffer wBuffer = new OakWBufferImpl(bb);
+        computer.accept(wBuffer);
+        unlockWrite(bb);
+        return SUCCESS;
+    }
+
+    /**
+     * Marks a value as DELETED and frees its slice (not including the header).
+     *
+     * @param s             the value Slice (including the header)
+     * @param memoryManager the memory manager to which the slice is returned to
+     * @param oldValue      in case of a conditional remove, this is the value to which the actual value is compared to
+     * @param transformer   value deserializer
+     * @param <V>           the type of the value
+     * @return {@code SUCCESS} if the value was removed successfully,
+     * {@code FAILURE} if the value is already deleted, or if it does not equal to {@code oldValue}
+     * {@code RETRY} if the value was moved.
```

Comment:
If the only meaning of "RETRY" is that the value was moved, can we name it "MOVED"? This will make some code using this return value more clear...